### ISSUE TITLE
Cache queries

### ISSUE THREAD

**Issue by jakebeal** [SynBioDex/tyto#21]
When TYTO performs a remote query, it takes several seconds.
While this is fine for occasional queries, it can make automation massively slow.
Can you please add caching for queries so that lookup failures can be resolved quickly after the first one?

**Comment by jakebeal** [comment]
Here's my workaround implementation that I'm using locally:
```
tyto_cache = {}
def tyto_lookup_with_caching(term: str) -> str:
    if term not in tyto_cache:
        try:
            tyto_cache[term] = tyto.SO.get_uri_by_term(term)
        except LookupError as e:
            tyto_cache[term] = e
    if isinstance(tyto_cache[term], LookupError):
        raise tyto_cache[term]
    else:
        return tyto_cache[term]
```

**Comment by bbartley** [comment]
Closed in #24
